### PR TITLE
781 - Fixed the forum auto suggest on forum archive page

### LINKED ISSUE #781: Forum search suggestion not working expected
**Describe the bug**
When you log in as a non-admin user and try to search forums using the search bar appearing on the forum page.
You will get the no results 'Sorry, there were no discussions found in suggestions 

**To Reproduce**

Enable the forum banner and search from theme settings >> forums (If not enabled).
Logged as a normal user and search for forums or discussions.
See error (Sorry, there were no discussions found).

**Expected behavior**
It should give result in suggestion

**Screenshots**

**Support ticket links**

https://www.loom.com/share/c10f53bfc5704fe0b96d5bd5dbdcf01e

### PR DESCRIPTION
**Pull request by janakkaneriya007**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #781 .

### How to test the changes in this Pull Request:

1. Enable the forum banner and search from theme settings >> forums (If not enabled).
2. Logged as a normal user and search for forums or discussions.
3. See the auto suggest results

### Proof Screenshots or Video
https://www.loom.com/share/7ddd640f37b542ebad8f842490049d87

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fixed the forum auto suggest on forum archive page

### CODE REVIEW

**Comment by mohit-buddyboss** [comment]
@janakkaneriya007 

Tested this after enabling the Forum banner and search option from Theme options. For the Admin Forum banner search and suggestions, functionality is working fine. But for Subscriber, it is not. 

**Issue:** If subscribers search anything using the search bar available on Forum banner then they doesn't get results for Private and Hidden groups discussion ( Hidden groups which logged in user is a member) on the search result page. Although, user is getting suggestions for Private and Hidden group discussion but not getting results on the search result page. The search result page shows only Public group discussions. 

For better understanding kindly see this video: https://www.loom.com/share/924e5d336aca4e87b8f6cee17d6628e0

cc @chetansatasiya @mehulkaklotar

**Comment by mehulkaklotar** [comment]
@KartikSuthar can you take over this PR? check the last comment https://github.com/buddyboss/buddyboss-platform/pull/814#issuecomment-654899913 and fix it and put it in review again.

**Comment by KartikSuthar** [comment]
@mohit-buddyboss @mehulkaklotar , 

As discussed in the scrum, We did not need to sync autocomplete with the forum search as both are coming from different components. We don't need to load autocomplete on the archive page while disabled forum searches from network settings:
http://prntscr.com/thdkyt

**Review by chetansatasiya — DISMISSED** [review]
@KartikSuthar @janakkaneriya007 👍 

@ankitguptaindia Need to test this.

**Review by mehulkaklotar — DISMISSED** [review]
Needs testing here. @ankitguptaindia

**Comment by mohit-buddyboss** [comment]
@KartikSuthar 

Tested on the branch `fix/781` of the Platform.

Issue 1: Under suggestions forum name is appearing but when user click on enter then on search result page Forum List not appears it shows only discussions. 
Expected: If suggestions shows Forum name then in search result Forum name also must appear.

Issue 2: When user search individual Forum then auto suggest shows it's name but when user search any Forum associated with Group then Auto suggest is not showing it's name in suggestion. For Public grp having forum also.

Issue 3 : Under suggestions Private grp discussions are appearing to Non grp members.
Expected : Private group discussions must not appear as a suggestions for Non Group Members.

Video: https://www.loom.com/share/9e1648c378154e0e942f36a95d650db4

**Comment by KartikSuthar** [comment]
@mohit-buddyboss 
Please review my comments as below.
> Tested on the branch `fix/781` of the Platform.
> 
> Issue 1: Under suggestions forum name is appearing but when user click on enter then on search result page Forum List not appears it shows only discussions.
> Expected: If suggestions shows Forum name then in search result Forum name also must appear.

I have checked and forum name is also coming there: http://prntscr.com/vemae8

> Issue 2: When user search individual Forum then auto suggest shows it's name but when user search any Forum associated with Group then Auto suggest is not showing it's name in suggestion. For Public grp having forum also.

At here autosuggestion comes from the network search component. When Groups search will be disabled from the backend it will show the group forums otherwise not: http://prntscr.com/vemi15

> Issue 3 : Under suggestions Private grp discussions are appearing to Non grp members.
> Expected : Private group discussions must not appear as a suggestions for Non Group Members.
> 
> Video: https://www.loom.com/share/9e1648c378154e0e942f36a95d650db4

I have fixed this issue.

File: src/bp-templates/bp-nouveau/includes/search/functions.php [hunk 1]
```diff
@@ -57,12 +57,22 @@ function bp_nouveau_search_enqueue_scripts() {
...
+		'forums_autocomplete'   => false,
 	);
 
 	if ( isset( $_GET["s"] ) ) {
 		$data["search_term"] = $_GET["s"];
 	}
 
+	if ( bp_is_active( 'forums' ) ) {
+		$data['forums_autocomplete'] = (
+			bp_is_search_post_type_enable( 'forum' ) ||
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/search/functions.php:69]:
```suggestion
			bp_is_search_post_type_enable( bbp_get_forum_post_type() ) ||
```
@janakkaneriya007 Use the function to get the dynamic value and it works always if someone change the slug.

File: src/bp-templates/bp-nouveau/includes/search/functions.php [hunk 2]
```diff
@@ -57,12 +57,22 @@ function bp_nouveau_search_enqueue_scripts() {
...
+			bp_is_search_post_type_enable( 'topic' ) ||
+			bp_is_search_post_type_taxonomy_enable( bbpress()->topic_tag_tax_id, 'topic' ) ||
+			bp_is_search_post_type_enable( 'reply' )
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/search/functions.php:72]: same here @janakkaneriya007

**Review by chetansatasiya — COMMENTED** [review]
Need to requested changes

**Review by chetansatasiya — APPROVED** [review]
Approved!

**Comment by KartikSuthar** [comment]
@mohit-buddyboss Can you please test this?

**Comment by sumitbuddyboss** [comment]
> @mohit-buddyboss
> Please review my comments as below.
> 
> > Tested on the branch `fix/781` of the Platform.
> > Issue 1: Under suggestions forum name is appearing but when user click on enter then on search result page Forum List not appears it shows only discussions.
> > Expected: If suggestions shows Forum name then in search result Forum name also must appear.
> 
> I have checked and forum name is also coming there: http://prntscr.com/vemae8
> 
> > Issue 2: When user search individual Forum then auto suggest shows it's name but when user search any Forum associated with Group then Auto suggest is not showing it's name in suggestion. For Public grp having forum also.
> 
> At here autosuggestion comes from the network search component. When Groups search will be disabled from the backend it will show the group forums otherwise not: http://prntscr.com/vemi15
> 
> > Issue 3 : Under suggestions Private grp discussions are appearing to Non grp members.
> > Expected : Private group discussions must not appear as a suggestions for Non Group Members.
> > Video: https://www.loom.com/share/9e1648c378154e0e942f36a95d650db4
> 
> I have fixed this issue.
> 
> Please retest now

Hello @KartikSuthar 
The related issue is resolved now. For proof please check the below video.
https://www.loom.com/share/2c789239d5d144b9b00e357e09178724

@chetansatasiya 
Ready for release 🚀 

cc @ankitguptaindia

**Review by sourabhmatolia — APPROVED** [review]

**Pull request merged**